lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3Input.java

```diff
@@ -82,47 +81,47 @@ public int readTail(byte[] buffer, int offset, int length)
         String range = "bytes=-%s".formatted(length);
         GetObjectRequest rangeRequest = request.toBuilder().range(range).build();
 
-        try (InputStream in = getObject(rangeRequest)) {
-            return readNBytes(in, buffer, offset, length);
-        }
-    }
-
-    @Override
-    public void close()
-    {
-        closed = true;
-    }
-
-    private void ensureOpen()
-            throws IOException
-    {
-        if (closed) {
-            throw new IOException("Input closed: " + location);
-        }
+        return read(buffer, offset, length, rangeRequest);
     }
 
-    private InputStream getObject(GetObjectRequest request)
+    private int read(byte[] buffer, int offset, int length, GetObjectRequest rangeRequest)
             throws IOException
     {
+        AtomicInteger result = new AtomicInteger(0);
```

Comment:
This shouldn't be needed, as the `getObject()` method can return the result directly from `ResponseTransformer`.

Reply:
🤦 I got fixated on it having to return an input stream. Done!